pymc_marketing/clv/distributions.py

```diff
@@ -152,36 +140,33 @@ class ContContractRV(RandomVariable):
     dtype = "floatX"
     _print_name = ("ContinuousContractual", "\\operatorname{ContinuousContractual}")
 
-    def make_node(self, rng, size, dtype, lam, p, T, T0):
+    def make_node(self, rng, size, dtype, lam, p, T):
         T = pt.as_tensor_variable(T)
-        T0 = pt.as_tensor_variable(T0)
 
-        return super().make_node(rng, size, dtype, lam, p, T, T0)
+        return super().make_node(rng, size, dtype, lam, p, T)
 
-    def __call__(self, lam, p, T, T0=0, size=None, **kwargs):
-        return super().__call__(lam, p, T, T0, size=size, **kwargs)
+    def __call__(self, lam, p, T, size=None, **kwargs):
+        return super().__call__(lam, p, T, size=size, **kwargs)
 
     @classmethod
-    def rng_fn(cls, rng, lam, p, T, T0, size) -> np.ndarray:
+    def rng_fn(cls, rng, lam, p, T, size) -> np.ndarray:
         size = pm.distributions.shape_utils.to_tuple(size)
 
         # To do: broadcast sizes
         lam = np.asarray(lam)
         p = np.asarray(p)
         T = np.asarray(T)
-        T0 = np.asarray(T0)
 
         if size == ():
-            size = np.broadcast_shapes(lam.shape, p.shape, T.shape, T0.shape)
+            size = np.broadcast_shapes(lam.shape, p.shape, T.shape)
 
         lam = np.broadcast_to(lam, size)
         p = np.broadcast_to(p, size)
         T = np.broadcast_to(T, size)
-        T0 = np.broadcast_to(T0, size)
 
         output = np.zeros(shape=size + (3,))
 
-        def sim_data(lam, p, T, T0):
```

Comment:
When I talked about vectorization in the original issue I was thinking about something slightly different.

The idea is to replace nested loop of this form:
```python
ys = []
# sim_data
for x in xs:
  y = y0
  while not cond(y):
    y = rng.foo(bar(y, x))
  ys.append(y)
return ys
```

By:
```python
ys = full(xs.shape, y0)
unfinished = ones(xs.shape, dtype=bool)
while any(unfinished):
  ys[unfinished] = rng.foo(bar(ys[unfinished], xs[unfinished])
  unfinished[unfinished] = not cond(unfinished[unfinished])
return ys
```

Where the repeated fancy indexing would be only done once.

This can be much faster because the calls to `foo` and `bar` can be numpy vectorized, at the cost of some overhead with fancy indexing, versus slow Python loops. 

It may not apply to all (any) of our distributions, and it's not super critical we investigate.